src/components/elements/table/header.js

```diff
@@ -1,21 +1,53 @@
+import { Component } from 'react'
 import cx from 'classnames'
 
-export default ({ api, property }) => {
-  if (!property.columns) {
-    return <thead/>
+export default class Header extends Component {
+  constructor(props) {
+    super()
+
+    this.state = {
+      sortedColumn: null,
+      direction: null,
+    }
+  }
+
+  render() {
+    if (!this.props.columns) {
+      return <thead />
+    }
+    return (
+      <thead>
+        <tr className='ct-table-header-row'>
+          {this.props.columns.map((column, columnIndex) => {
+            return (
+              <th
+                onClick={e => this.onSort(e, columnIndex) }
+                key={column.id}
+                className='ct-table-header'>
+                <a href='#'
+                  className={'ct-table-header-sort-link'}>
+                  {column.title}</a>
+                <span className={cx('ct-table-header-sort-direction', {
+                  'ct-table-header-sort-ascending': this.state.sortedColumn == columnIndex &&
+                    !this.state.direction,
+                  'ct-table-header-sort-descending': this.state.sortedColumn == columnIndex &&
+                    this.state.direction})}>
+                </span>
+              </th>
+            )
+          })
+          }
+        </tr>
+      </thead>
+    )
+  }
+
+  onSort(e, columnIndex) {
+    e.preventDefault()
+    this.setState({
```

Comment:
I'm not sure if you think it would be a good idea or not but maybe we could keep all sorting state at the level above this one (`index.js`) and send the sorted column index and direction as properties to this component. Then we could keep this component stateless.